apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java

```diff
@@ -92,6 +98,23 @@ public void shutdown() {
     @Override
     public void run() {
         LOGGER.debug("Selected collector grpc service running, reconnect:{}.", reconnect);
+        if (IS_RESOLVE_DNS_PERIODICALLY || reconnect) {
+            String backendService = Config.Collector.BACKEND_SERVICE.split(",")[0];
+            try {
+                String[] domainAndPort = backendService.split(":");
+
+                List<String> newGrpcServers = Arrays
+                        .stream(InetAddress.getAllByName(domainAndPort[0]))
+                        .map(InetAddress::getHostAddress)
+                        .map(ip -> String.format("%s:%s", ip, domainAndPort[1]))
+                        .collect(Collectors.toList());
+
+                grpcServers = newGrpcServers;
```

Comment:
You are overriding the `grpcServers`, you wouldn't have the chance to resolve DNS again.

Reply:
I don't get it

Reply:
Oh, sorry, my mistake. You split the config string every time.